### PR TITLE
Clean build dir

### LINKED ISSUE #89: Fix/Clean the build directory
I think the whole build folder can be removed. This would make it easier to work with during development as "build" is often used as the default folder for generated build files. But it might make sense to actually clean it up a bit.

@mat007 Could you check what build files are actually required?

- The VC project files can surely go, CMake creates them and they are (likely) outdated anyway. Done that in my PR but can extract
- What are the build.xml and build.properties files? Is that some other IDE? Are those still correct and/or required or can they be removed? I'd suggest to only use CMake and let that create the files for whatever IDE you'd like to use. Happy to help here if required, but I think the current CMake build system is pretty complete already
- The Dockerfiles might be nice, but I guess those packages could just be mentioned in the Readme. If you'd really like to keep them, maybe put them in another folder. (e.g. "scripts", see below)
- The build.[sh,bat] might be better suited for a "scripts" folder
- The version.hpp is a good idea. This can be moved to the root as version.hpp.cmake and configured by CMake and added to the install procedure.

### PR DESCRIPTION
**Pull request by Flamefire**
This closes #89 

See the issue and individual commits for details

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+jobs:
+  release:
+    name: Create Release
+    runs-on: ubuntu-latest
+    env:
+      BOOST_VERSION: 1.71.0
+      BOOST_ROOT: ${{github.workspace}}/dependencies/boost
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/checkout@v2
```
Comment (mat007) [inline .github/workflows/release.yml:21]: Why does this need `checkout` twice?

File: build/build.sh [hunk 2]
```diff
@@ -21,13 +14,7 @@ cd "$BOOST"
...
-copy -r "$BOOST"/tools/boostbook/dtd "$BOOSTBOOK_DIR"
-copy -r build/boostbook/* "$BOOSTBOOK_DIR"
-copy "$BOOST"/doc/src/boostbook.css doc/html
-copy "$BOOST"/doc/src/images/*.png doc/html/images
-copy "$BOOST"/doc/src/images/callouts/*.png doc/html/images/callouts
+scripts/build_doc.sh "$@"
 
 cd "$BOOST"
-./b2 "$PROJECT_DIR/doc" -q "$@"
+./b2 "$PROJECT_DIR/doc//mock_examples" -q "$@"
```
Comment (mat007) [inline build/build.sh]: Why the two `/`?

File: .github/workflows/release.yml (hunk 1 above)
Comment (Flamefire) [reply .github/workflows/release.yml:21]: The first is this project the 2nd is boost. See the with part below

**Comment by Flamefire** [comment]
Sure. I created one in my fork: https://github.com/Flamefire/turtle/releases/tag/v0.0.1

File: build/build.sh (hunk 2 above)
Comment (Flamefire) [reply build/build.sh]: That's b2: go into that folder and build that target

Comment (mat007) [reply build/build.sh]: I meant why `"$PROJECT_DIR/doc//mock_examples"` instead of `"$PROJECT_DIR/doc/mock_examples"`?

Comment (Flamefire) [reply build/build.sh]: B2 uses a double slash to delimit folder and target 

mock_examples is a target in the folder $PROJECT_DIR/doc. See the jamfile

Comment (mat007) [reply build/build.sh]: Oh right, thanks!

**Pull request merged**